### PR TITLE
Add sysl.arrai to polish Sysl models for use in arr.ai

### PR DESCRIPTION
**Pull request by orlade-anz**

### CODE REVIEW

File: pkg/arrai/sysl.arrai [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        cond {
+            .@item.ret?:{}: tweak(.),
+            _: .,
+        }
+;
+
+# Applies polish functions to the appropriate elements of a Sysl model.
+let polish = \m
+    polish_returns(m)
+;
```
Comment (nofun97) [inline pkg/arrai/sysl.arrai]: this seems unnecessary, why not directly use `polish_returns`?

File: pkg/arrai/sysl.arrai [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+# Export the functions from the module.
+let export = (
+    :parse_return,
+    :polish,
+
```
Comment (nofun97) [inline pkg/arrai/sysl.arrai]: remove newline

File: pkg/arrai/sysl.arrai [hunk 3]
```diff
@@ -0,0 +1,43 @@
+# Performs transformations of Sysl models, producing "better" Sysl models.
+
+# Invokes a macro on a string as if it were source code at parsing time.
+let invokeMacro = \macro \s
+    macro -> (//dict(.@transform) >>> \rule \fn fn(//grammar.parse(.@grammar, rule, s))).@value;
```
Comment (nofun97) [inline pkg/arrai/sysl.arrai]: you can iterate through tuples with the `:>` operator. For example:

```
@> (a: 1, b: 2, c: 3) :> . + 1
(a: 2, b: 3, c: 4)
```

Comment (orlade-anz) [reply pkg/arrai/sysl.arrai]: Indeed, but I need the name of the tuple attribute in order to compute its new value, not just the current value.

File: pkg/arrai/sysl.arrai (hunk 1 above)
Comment (orlade-anz) [reply pkg/arrai/sysl.arrai]: Added a TODO to add more polish :). It's just a placeholder.

File: pkg/arrai/sysl.arrai (hunk 2 above)
Comment (orlade-anz) [reply pkg/arrai/sysl.arrai]: It was separating sysl func from util funcs. Moved util funcs to util.arrai.

**Comment by orlade-anz** [comment]
Fixes https://github.com/arr-ai/arrai/issues/510.

**Pull request merged**